homeassistant/components/mqtt/switch.py

```diff
@@ -126,7 +128,12 @@ def _setup_from_config(self, config):
         state_off = config.get(CONF_STATE_OFF)
         self._state_off = state_off if state_off else config[CONF_PAYLOAD_OFF]
 
-        self._optimistic = config[CONF_OPTIMISTIC]
+        self._optimistic = (
+            config[CONF_OPTIMISTIC] or config.get(CONF_STATE_TOPIC) is None
+        )
+
+        if self._optimistic:
+            self._state = False
```

Comment:
Why should an optimistic switch not be initialized to unknown?

Reply:
Because an unknown state is not supported in this scenario. Via the gui you can only set off or on. We are not processing the device's state and assume optimistic that we know what the state is. So why break the current functionallity then.
On the other hand, an unkown current state could be acceptable if the state is restored after a reload, it will only be unkown the very first time. So let me know if I should change the inital state here.

Reply:
Frontend is changed here to handle "unknown" for toggle entities: <https://github.com/home-assistant/frontend/pull/11522>, I think we can drop this change here.

Reply:
ok.